webapp/src/components/App/Singlestudy/HomeView/InformationView/LauncherHistory/JobStepper.tsx

```diff
@@ -68,6 +79,9 @@ export default function VerticalLinearStepper(props: Props) {
   const [openConfirmationDialog, setOpenConfirmationDialog] =
     useState<boolean>(false);
   const [jobIdKill, setJobIdKill] = useState<string>();
+  const [openDigestDialog, setOpenDigestDialog] = useState<LaunchJob | null>(
+    null,
+  );
```

Comment:
Instead of using a single state to manage both the visibility and the data of the dialog, you should use two states: one for controlling the visibility and another for storing the data to be passed to the dialog.

```typescript
 const [openDigestDialog, setOpenDigestDialog] = useState(false);
 const [selectedJob, setSelectedJob] = useState<LaunchJob | null>(null);
```
then adapt the rest:

```typescript
// add the event handler
  const handleOpenDigestDialog = (job: LaunchJob) => {
    setSelectedJob(job);
    setOpenDigestDialog(true);
  };
  
  // use it here
  {job.status === "success" && (
      <Tooltip title="Digest">
        <EqualizerIcon
           onClick={handleOpenDigestDialog}
            sx={iconSx}
             />
        </Tooltip>
  )}
```

adapt this too/
```typescript
{openDigestDialog && selectedJob && (
        <DigestDialog
          open
          studyId={selectedJob.studyId}
          outputId={selectedJob.outputId}
          onOk={() => setOpenDigestDialog(false)}
        />
      )}
```

Reply:
I use an easy way because the view must be redesign, so I wanted to change few codes.

The good way would to have a common state for all the dialog like this for example:

```typescript
const [dialogState, setDialogState] = useState<{ type?: string; data?: LaunchJob }>({});
```

Two states for this is not necessary.

Reply:
I wanted to change few codes because the view must be redesigned, but I will make a change.